### ISSUE TITLE
[fixed] psych 4.0.4 breaks rubygems on ruby 2.5 on windows (with default rubygems version 2.7.6.3)

### ISSUE THREAD

**Issue by hurricup** [rubygems/rubygems#5547]
Not sure if this is a psych or rubygems issue, but after installing most recent `psych` attempt to install gem with any additional options causes:
```
c:/Ruby25-x64/bin/ruby c:/ruby25-x64/bin/gem update --system
Updating rubygems-update
Fetching: rubygems-update-3.3.13.gem (100%)
ERROR:  While executing gem ... (ArgumentError)
    wrong number of arguments (given 4, expected 1)
```
1. Removed `psych`
2. Was able to run `gem update`
3. Installed `psych` back
4. ~~Problem is still there, with something like `gem install gemname --user-install`~~ (Upd: it works. Not sure why failed first time. Feel free to close then, consider just FYI). 

Observing this only on 2.5

```
RubyGems Environment: 
  - RUBYGEMS VERSION: 3.3.13
  - RUBY VERSION: 2.5.9 (2021-04-05 patchlevel 229) [x64-mingw32]
  - INSTALLATION DIRECTORY: c:/Ruby25-x64/lib/ruby/gems/2.5.0
  - USER INSTALLATION DIRECTORY: C:/Users/builduser/.gem/ruby/2.5.0
  - RUBY EXECUTABLE: c:/Ruby25-x64/bin/ruby.exe
  - GIT EXECUTABLE: C:\Program Files\Git\cmd/git.EXE
  - EXECUTABLE DIRECTORY: c:/Ruby25-x64/bin
  - SPEC CACHE DIRECTORY: C:/Users/builduser/.gem/specs
  - SYSTEM CONFIGURATION DIRECTORY: C:/ProgramData
  - RUBYGEMS PLATFORMS:
     - ruby
     - x64-mingw32
  - GEM PATHS:
     - c:/Ruby25-x64/lib/ruby/gems/2.5.0
     - C:/Users/builduser/.gem/ruby/2.5.0
  - GEM CONFIGURATION:
     - :update_sources => true
     - :verbose => true
     - :backtrace => false
     - :bulk_threshold => 1000
  - REMOTE SOURCES:
     - https://rubygems.org/
```

**Comment by deivid-rodriguez** [comment]
Hi, yes, `Psych.load` completed removed the interface based on positional arguments, in favor of keyword parameters. And RubyGems 2.7.6.3 was using the old Psych 3 way. This was fixed in [RubyGems 3.1](https://github.com/rubygems/rubygems/commit/b7995ca4ea8452b4511f4afa7eddd5374fca21fc).

**Comment by deivid-rodriguez** [comment]
I will close this, it was a bit of an unfortunate situation but it's been fixed for a while. Updating RubyGems and/or Ruby should do the trick!
